lib/vast/Dialect/HighLevel/HighLevelOps.cpp

```diff
@@ -231,10 +231,32 @@ namespace vast::hl
         return (*this)->getOperand(0);
     }
 
+    void build_logic_op(Builder &bld, State &st, Type type, BuilderCallback lhs, BuilderCallback rhs)
+    {
+        VAST_ASSERT(lhs && "the builder callback for 'lhs' region must be present");
```

Comment:
Shouldn't this condition be present in the API? (take a type that will require callback to be there)

Reply:
Also it is copy-paste from if-op, where else branch might be missing. This was catching bugs where you misused API.
But in the case of binary op you require both callbacks.